File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/android/src/main/kotlin/com/danhdue/{{name.snakeCase()}}/presentation/{{name.pascalCase()}}PlatformView.kt
/*
 * Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.
 */

/*
 * coverage:ignore-file
 */

package com.danhdue.{{name.snakeCase()}}.presentation

import android.content.Context
import android.view.View
import androidx.compose.ui.platform.ComposeView
import io.flutter.plugin.platform.PlatformView

class {{name.pascalCase()}}PlatformView(
    context: Context,
    viewId: Int,
    args: Any?,
    viewModel: {{name.pascalCase()}}ViewModel
) : PlatformView {
    private val composeView = ComposeView(context).apply {
        setContent {
            {{name.pascalCase()}}Screen(viewModel = viewModel)
        }
    }

    override fun getView(): View = composeView

    override fun dispose() {}
}

File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/ios/{{name.snakeCase()}}/Sources/{{name.snakeCase()}}/Presentation/{{name.pascalCase()}}PlatformView.swift
// Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.

// coverage:ignore-file

import Flutter
import SwiftUI
import UIKit

public class {{name.pascalCase()}}PlatformView: NSObject, FlutterPlatformView {
    private let hostingController: UIHostingController<{{name.pascalCase()}}View>

    public init(
        frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?,
        viewModel: {{name.pascalCase()}}ViewModel
    ) {
        let controller = UIHostingController(rootView: {{name.pascalCase()}}View(viewModel: viewModel))
        controller.view.frame = frame
        controller.view.backgroundColor = .clear
        self.hostingController = controller
        super.init()
    }

    public func view() -> UIView {
        hostingController.view
    }
}
